src/components/dialogs/contingency-list/contingency-list-utils.js

```diff
@@ -0,0 +1,207 @@
+import {
+    getRangeInputEmptyDataForm,
+    getRangeInputSchema,
+} from '../../utils/range-input';
+
+export const DEFAULT_TABLE_ROWS = [{}, {}, {}];
+
+export const CONTINGENCY_LIST_EQUIPMENTS = {
+    LINE: { id: 'LINE', label: 'Lines' },
+    TWO_WINDINGS_TRANSFORMER: {
+        id: 'TWO_WINDINGS_TRANSFORMER',
+        label: 'TwoWindingsTransformers',
+    },
+    GENERATOR: { id: 'GENERATOR', label: 'Generators' },
+    SHUNT_COMPENSATOR: { id: 'SHUNT_COMPENSATOR', label: 'ShuntCompensators' },
+    HVDC_LINE: { id: 'HVDC_LINE', label: 'HvdcLines' },
+    BUSBAR_SECTION: { id: 'BUSBAR_SECTION', label: 'BusBarSections' },
+    DANGLING_LINE: { id: 'DANGLING_LINE', label: 'DanglingLines' },
+};
+const checkNameIsUnique = (name, activeDirectory) => {
+    if (!name) {
+        return false;
+    }
+    return new Promise((resolve) => {
+        elementExists(activeDirectory, name, ElementType.CONTINGENCY_LIST).then(
+            (val) => resolve(!val)
+        );
+    });
+};
+
+export const getSchema = (activeDirectory) => {
+    return yup.object().shape({
+        [NAME]: yup
+            .string()
+            .nullable()
+            .when([EQUIPMENT_ID], {
+                is: null,
+                then: (schema) =>
+                    schema
+                        .required('nameEmpty')
+                        .test('checkIfUniqueName', 'nameAlreadyUsed', (name) =>
+                            checkNameIsUnique(name, activeDirectory)
+                        ),
+            }),
+        [EQUIPMENT_TYPE]: yup.string().nullable(),
+        [CONTINGENCY_LIST_TYPE]: yup.string().nullable(),
+        [EQUIPMENT_TABLE]: yup.array().of(
+            yup.object().shape({
+                [CONTINGENCY_NAME]: yup.string().nullable(),
+                [EQUIPMENT_IDS]: yup.array().of(yup.string().nullable()),
+            })
+        ),
+        [SCRIPT]: yup.string().nullable(),
+        [COUNTRIES_1]: yup.array().of(yup.string().nullable()),
+        [COUNTRIES_2]: yup.array().of(yup.string().nullable()),
+        ...getRangeInputSchema(NOMINAL_VOLTAGE_1),
+        ...getRangeInputSchema(NOMINAL_VOLTAGE_2),
+    });
+};
+
+export const getEmptyFormData = () => ({
+    [EQUIPMENT_ID]: null,
+    [EQUIPMENT_TABLE]: DEFAULT_TABLE_ROWS,
+    [CONTINGENCY_LIST_TYPE]: ContingencyListTypeRefactor.CRITERIA_BASED.id,
+    [EQUIPMENT_TYPE]: '',
+    [COUNTRIES_1]: [],
+    [COUNTRIES_2]: [],
+    [SCRIPT]: null,
+    ...getRangeInputEmptyDataForm(NOMINAL_VOLTAGE_1),
+    ...getRangeInputEmptyDataForm(NOMINAL_VOLTAGE_2),
+});
+
+export const getFormDataFromFetchedElement = (
+    response,
+    contingencyListType,
+    contingencyListId,
+    name
+) => {
+    switch (contingencyListType) {
+        case ContingencyListTypeRefactor.CRITERIA_BASED.id:
+            return {
+                [NAME]: name,
+                [EQUIPMENT_TYPE]: response?.equipmentType,
+                [COUNTRIES_1]: response?.countries1,
```

Comment:
I don't know how it's done elsewhere, but for me, we should only use ? when the field is optional. In this case, I doubt that countries 1 is optional and not countries 2.